smarts/env/hiway_env.py

```diff
@@ -110,22 +114,46 @@ def __init__(
                 "timestep_sec has been deprecated in favor of fixed_timestep_sec.  Please update your code.",
                 category=DeprecationWarning,
             )
+        if agent_specs is not None:
+            warnings.warn(
+                "agent_specs has been deprecated in favor of agent_interfaces.  Please update your code.",
+                category=DeprecationWarning,
+            )
         if not fixed_timestep_sec:
             fixed_timestep_sec = timestep_sec or 0.1
 
-        self._agent_specs = agent_specs
+        def assert_type_and_return(v, types, target):
+            if not isinstance(v, types):
+                raise TypeError(
+                    f"{target}must be supplied with only {target} in values."
+                )
+            return v
```

Comment:
Is this type check needed or can it be removed?

Reply:
It is not really needed but was used to ensure that if someone is using `agent_specs` that they are not using `None` in the optional `AgentSpec.agent_interface`. I suppose we are deprecating `agent_spec` anyway so I will remove this check.